backend/src/apiserver/server/api_converter.go

```diff
@@ -134,6 +141,14 @@ func toApiParameters(paramsString string) ([]*api.Parameter, error) {
 		var value string
 		if param.Value != nil {
 			value = *param.Value
+			// Patch the GCS default values if available
+			if common.GetBoolConfigWithDefault(HasDefaultBucketEnvVar, false) {
+				var err error
+				value, err = PatchPipelineDefaultParameter(value)
+				if err != nil {
+					return nil, fmt.Errorf("failed to patch default value to pipeline. Error: %v", err)
```

Comment:
Maybe return NewInternalServerError to be consistent?

Reply:
Sync with Yang, we decide to remove this part.

The actual replacement occurs in CreateRun. This part is only for informational purpose (user can get the actual parameter value from create run UI).